### DISCUSSION TITLE
Error with Elasticsearch container

### DISCUSSION THREAD

**Discussion by bnouyou** [broadinstitute/seqr#2435] [Q&A]
Hello,

I have installed seqr as local instance. And it worked well.
Today I restarted the docker containers, and I suppose it pulls the last version of the code, and maybe something has changed with Elasticsearch. Now my elasticsearch container doesn't start anymore. And I have this error on logs:

```
elasticsearch      | java.lang.OutOfMemoryError: Java heap space
elasticsearch      | Dumping heap to data/java_pid6.hprof ...
elasticsearch      | Unable to create data/java_pid6.hprof: File exists
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Connection evictor"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[scheduler][T#1]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][scheduler][T#1]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "ticker-schedule-trigger-engine"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][generic][T#13]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][warmer][T#2]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][fetch_shard_started][T#4]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][warmer][T#1]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[scheduler][T#1]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][snapshot][T#2]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][management][T#4]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][generic][T#7]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][http_server_worker][T#1]"
elasticsearch      |
elasticsearch      | Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "elasticsearch[def8de833659][refresh][T#1]"
elasticsearch exited with code 127
```

This is what I have in the web interface:

![image](https://user-images.githubusercontent.com/22322228/151802264-7834059f-16c3-460e-b2e6-a14a528d302a.png)

Edit: actually, maybe it hasen't pulled the last version of the code, I haven't changed the docker-compose file. And the error seems related to JVM memory problem for elasticsearch.

Thank you in advance for your help.

**Comment by hanars** [comment]
Elasticsearch does not pull the latest version of code, so I agree that thats not the issue. You're seeing out of memory issues, which means you have not allocated sufficient resources to the JVM. This could be for a few different reasons. Can you run the following and let me know the output:
```
docker-compose exec elasticsearch grep vm.max_map_count /etc/sysctl.conf
```

Also, can you go to docker Preferences > Resources > Advanced and check what you have the memory limit set to?

**Reply by bnouyou** [reply]
Hello,

I did:
```
docker-compose exec elasticsearch grep vm.max_map_count /etc/sysctl.conf 
```
And I have:
```
vm.max_map_count=262144
```

I am using docker in command line in Linux, I don't know how to check the memory limite as you write in Preferences > Resources > Advanced.


Actually, I seached how to solve JVM issues, and I found something: I have added this line in the docker-compose file for elasticsearch container, and it seems working now:

```
environment:
      - ES_JAVA_OPTS=-Xms4g -Xmx4g
```

How much memory should we allow to elasticsearch to make it working ?

**Reply by hanars** [reply]
How much memory you need is very proportional to how much data you have in elasticsearch. We have around 11Tb of data in elasticsearch and allocate 22g for memory, but if 4g is working for you now that should be fine. You can also go to the `<seqr_host>/data_management/elasticsearch_status` page in seqr which will show you details of you elasticsearch cluster, including what percent of the heap is being used by each disk. If any of them are regularly using over 85% of their available heap, you may want to allocate more memory

**Reply by bnouyou** [reply]
I don't have as much data as you do, so I think 4g is enough for me for now. I will track it in the future, and if I have the same error again, I will know how to solve it. 
Thank you for your help.
